src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password)
+            throws AuthenticationException {
+        Optional<Driver> driver = driverService.findDriverByLogin(login);
```

Comment:
It is not `driver` but `driver` wrapped with `Optional`, please give a proper name